File: logic/src/main/java/com/tp/holdem/logic/model/PlayerNumber.java
package com.tp.holdem.logic.model;

import lombok.Value;

@Value(staticConstructor = "of")
public class PlayerNumber {
	private Integer number;

	public static PlayerNumber empty() {
		return new PlayerNumber(null);
	}

	public boolean exists() {
		return number != null;
	}
}

File: logic/src/main/kotlin/com/tp/holdem/logic/model/PlayerNumber.kt
package com.tp.holdem.logic.model

data class PlayerNumber(val number: Int = -1) {
    companion object {
        fun empty(): PlayerNumber {
            return PlayerNumber(-1)
        }

        fun of(number: Int): PlayerNumber {
            return PlayerNumber(number)
        }
    }

    fun exists(): Boolean {
        return number != -1
    }
}
